src/Bicep.Core/Utils/Environment.cs

```diff
@@ -6,4 +6,7 @@ public class Environment : IEnvironment
 {
     public string? GetVariable(string variable)
         => System.Environment.GetEnvironmentVariable(variable);
+
+    public string[] GetVariableNames()
+        => System.Environment.GetEnvironmentVariables().Keys.Cast<string>().ToArray();
```

Comment:
nit: We can just return `IEnumerable<string>` without calling `ToArray()` to avoid array allocation.

Reply:
Absolutely! Fixed.